vue-app/src/components/RecipientSubmissionWidget.vue

```diff
@@ -165,67 +169,120 @@ export default class RecipientSubmissionWidget extends Vue {
   }
 
   get canSubmit(): boolean {
-    return !!this.currentUser && !!this.walletProvider && !!this.registryInfo
+    return (
+      !!this.currentUser &&
+      !!this.walletProvider &&
+      !!this.$store.state.recipientRegistryInfo
+    )
   }
 
   get hasTxError(): boolean {
     return !!this.txError
   }
 
   get txHasDeposit(): boolean {
-    return !!this.registryInfo?.deposit
+    return !!this.$store.state.recipientRegistryInfo?.deposit
   }
 
   get depositAmount(): string {
-    return this.registryInfo
-      ? formatAmount(this.registryInfo.deposit, 18)
+    return this.$store.state.recipientRegistryInfo
+      ? formatAmount(this.$store.state.recipientRegistryInfo.deposit, 18)
       : '...'
   }
 
+  get gasFeeAmount(): string {
+    return this.estimatedGasFee ? formatAmount(this.estimatedGasFee, 18) : '-'
+  }
+
   get depositToken(): string {
-    return this.registryInfo?.depositToken ?? ''
+    return this.$store.state.recipientRegistryInfo?.depositToken ?? ''
   }
 
   handleSubmit(): void {
     this.addRecipient()
   }
 
+  public calculateFiatFee(ethAmount: BigNumber): string {
+    if (this.$store.state.recipientRegistryInfo && this.ethPrice) {
+      return Number(
+        this.ethPrice.ethereum.usd * Number(formatAmount(ethAmount, 18))
+      ).toFixed(2)
+    }
+    return '-'
+  }
+
+  private async getEstimatedGasFee(): Promise<BigNumber> {
+    const { recipient, recipientRegistryAddress, recipientRegistryInfo } =
+      this.$store.state
+
+    if (
+      recipientRegistryInfo &&
+      this.ethPrice &&
+      this.walletProvider &&
+      recipientRegistryAddress &&
+      recipient
+    ) {
+      const registry = new Contract(
+        recipientRegistryAddress,
+        OptimisticRecipientRegistry,
+        this.walletProvider
+      )
+
+      return await registry.estimateGas.addRecipient(
+        recipient.fund.address,
+        JSON.stringify({
```

Comment:
I think we want to use `formToRecipientData` from /api/recipient-registry-optimistic.ts here. See how we construct it here: https://github.com/ethereum/clrfund/blob/develop/vue-app/src/api/recipient-registry-optimistic.ts#L279